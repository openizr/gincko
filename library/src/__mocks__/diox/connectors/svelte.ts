/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { readable } from 'svelte/store/index';

/**
 * diox/connectors/svelte mock.
 */
const connect = vi.fn(() => {
  const state = {
    steps: [{ id: 'start' }, { id: 'end' }],
    loading: process.env.LOADING === 'true',
  };
  return vi.fn(() => readable(state, (set) => {
    set(state);
  }));
});

export default connect;
