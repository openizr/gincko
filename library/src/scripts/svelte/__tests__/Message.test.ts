/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { render } from '@testing-library/svelte';
import Message from 'scripts/svelte/Message.svelte';

vi.mock('biuty/svelte');

describe('svelte/Message', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - default props', async () => {
    const { container, rerender } = render(Message, { props: { id: 'test', label: '*Test*' } });
    expect(container.firstChild).toMatchSnapshot();
    rerender({ props: { id: 'test', label: '*Test*', modifiers: null as unknown as undefined } });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - custom props', async () => {
    const { container } = render(Message, { props: { id: 'test', label: '*Test*', modifiers: 'strong' } });
    expect(container.firstChild).toMatchSnapshot();
  });
});
