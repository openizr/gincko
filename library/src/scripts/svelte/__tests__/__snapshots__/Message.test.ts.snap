// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`svelte/Message > renders correctly - custom props 1`] = `
<div>
  <section
    class="ui-message strong"
    id="test"
  >
    MARKDOWN FOR *Test*, false
  </section>
  <!--&lt;Message&gt;-->
</div>
`;

exports[`svelte/Message > renders correctly - default props 1`] = `
<div>
  <section
    class="ui-message "
    id="test"
  >
    MARKDOWN FOR *Test*, false
  </section>
  <!--&lt;Message&gt;-->
</div>
`;

exports[`svelte/Message > renders correctly - default props 2`] = `
<div>
  <section
    class="ui-message "
    id="test"
  >
    MARKDOWN FOR *Test*, false
  </section>
  <!--&lt;Message&gt;-->
</div>
`;
